src/main/java/uk/gov/ons/census/casesvc/service/FulfilmentRequestProcessor.java

```diff
@@ -22,44 +28,110 @@
+              HOUSEHOLD_INDIVIDUAL_RESPONSE_REQUEST_WALES_ENGLISH,
+              HOUSEHOLD_INDIVIDUAL_RESPONSE_REQUEST_WALES_WELSH,
+              HOUSEHOLD_INDIVIDUAL_RESPONSE_REQUEST_NORTHERN_IRELAND));
 
   private final CaseRepository caseRepository;
   private final EventLogger eventLogger;
+  private final CaseProcessor caseProcessor;
 
-  public FulfilmentRequestProcessor(CaseRepository caseRepository, EventLogger eventLogger) {
+  public FulfilmentRequestProcessor(
+      CaseRepository caseRepository, EventLogger eventLogger, CaseProcessor caseProcessor) {
     this.caseRepository = caseRepository;
     this.eventLogger = eventLogger;
+    this.caseProcessor = caseProcessor;
   }
 
   public void processFulfilmentRequest(ResponseManagementEvent fulfilmentRequest) {
     EventDTO fulfilmentRequestEvent = fulfilmentRequest.getEvent();
     FulfilmentRequestDTO fulfilmentRequestPayload =
         fulfilmentRequest.getPayload().getFulfilmentRequest();
 
-    String caseId = fulfilmentRequestPayload.getCaseId();
+    Case caze = getCaseByCaseId(fulfilmentRequestPayload);
 
-    Optional<Case> cazeResult = caseRepository.findByCaseId(UUID.fromString(caseId));
+    validateFulfilmentRequiredFields(
+        fulfilmentRequestEvent, fulfilmentRequestPayload, caze.getCaseId().toString());
 
-    if (cazeResult.isEmpty()) {
-      log.error(CASE_NOT_FOUND_ERROR);
-      throw new RuntimeException(String.format("Case ID '%s' not found!", caseId));
+    eventLogger.logFulfilmentRequestedEvent(
+        caze,
+        caze.getCaseId(),
+        fulfilmentRequestEvent.getDateTime(),
+        FULFILMENT_REQUEST_RECEIVED,
+        FULFILMENT_REQUESTED,
+        fulfilmentRequestPayload,
+        fulfilmentRequestEvent);
+
+    if (individualResponseRequestCodes.contains(fulfilmentRequestPayload.getFulfilmentCode())) {
+      Case individualResponseCase = saveIndividualResponseCase(caze);
+      caseProcessor.emitCaseCreatedEvent(individualResponseCase);
     }
+  }
+
+  private Case saveIndividualResponseCase(Case caze) {
+    Case individualResponseCase =
+        caze.toBuilder()
+            .caseId(UUID.randomUUID())
+            .caseRef(caseProcessor.getUniqueCaseRef())
+            .uacQidLinks(null)
+            .events(null)
+            .createdDateTime(OffsetDateTime.now())
+            .state(CaseState.ACTIONABLE)
+            .receiptReceived(false)
+            .refusalReceived(false)
+            .addressType(HOUSEHOLD_INDIVIDUAL_RESPONSE_ADDRESS_TYPE)
+            .addressLevel(null)
+            .htcWillingness(null)
+            .htcDigital(null)
+            .fieldCoordinatorId(null)
+            .fieldOfficerId(null)
+            .treatmentCode(null)
+            .ceExpectedCapacity(null)
+            .build();
+
+    caseRepository.save(individualResponseCase);
+
+    return individualResponseCase;
+  }
 
-    if (fulfilmentRequestEvent.getDateTime() == null) {
+  private void validateFulfilmentRequiredFields(
```

Comment:
if we're going to start building validators we're going to be working until the end of time producing YAGNI code instead of delivering features and functionality. What is the requirement for this?

Reply:
validators removed